hibernate-rx-core/src/test/java/org/hibernate/rx/configuration/RxConnectionPoolTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.rx.configuration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.rx.cfg.AvailableRxSettings;
+import org.hibernate.rx.containers.PostgreSQLDatabase;
+import org.hibernate.rx.service.RxConnectionPoolProviderImpl;
+import org.hibernate.rx.service.initiator.RxConnectionPoolProvider;
+import org.junit.Test;
+
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.Pool;
+
+public class RxConnectionPoolTest {
+
+	@Test
+	public void configureWithPool() {
+		String url = PostgreSQLDatabase.getJdbcUrl().substring( "jdbc:".length() );
+		Pool pgPool = PgPool.pool( url );
+		Map<String,Object> config = new HashMap<>();
+		config.put( AvailableRxSettings.VERTX_POOL, pgPool );
+		RxConnectionPoolProvider rxPool = new RxConnectionPoolProviderImpl( config );
+		rxPool.getConnection().close();
+	}
+	
+	@Test
+	public void configureWithJdbcUrl() {
+		String url = PostgreSQLDatabase.getJdbcUrl();
+		Map<String,Object> config = new HashMap<>();
+		config.put( AvailableSettings.URL, url );
+		RxConnectionPoolProvider rxPool = new RxConnectionPoolProviderImpl( config );
+		rxPool.getConnection().close();
+	}
+	
+	@Test
+	public void configureWithCredentials() {
+		// Set up URL with invalid credentials so we can ensure settings are used
```

Comment:
Maybe I'm tired but I don't understand the logic here. If the credentials are wrong, shouldn't the test fail somewhere?

Reply:
The intent of this test is to ensure that explicitly defined credentials (e.g. `config.put( AvailableSettings.USER, "admin" )`) take precedence over any user/password that might be part of the JDBC URL. In this test we purposely put bad values in the JDBC URL, but good values in the config map, and expect the connection to succeed which would prove the config map credentials take precedence.